### PR TITLE
CAF-2696: Integration testing

### PR DESCRIPTION
**Pull request by CJComac**
Based upon Connor's branch.

ElasticAuditTestCaseValidationIT tests the generated AuditLog library methods with test event data.

### CODE REVIEW

File: caf-audit/src/test/java/com/hpe/caf/auditing/elastic/ElasticAuditIT.java [hunk 1]
```diff
@@ -328,38 +328,94 @@ public void testMultiTenantESIndexing() throws Exception
...
 
-                SearchHit[] tenant2IndexHits = new SearchHit[0];
+                //  Delete test indexes after verification is complete.
+                deleteTenantIndices(transportClient, tenantIndexIds);
+            }
+        }
+    }
 
-                String tenant2IndexId = ES_INDEX_PREFIX + tenant2Id;
+    private void deleteTenantIndices(TransportClient client, String[] indices) {
```
Comment (mulhollandc) [inline caf-audit/src/test/java/com/hpe/caf/auditing/elastic/ElasticAuditIT.java]: IMO I would rename this method to deleteIndex, deleteIndexes or deleteIndices. Drop the tenant bit.

File: caf-audit/src/test/java/com/hpe/caf/auditing/elastic/ElasticAuditIT.java [hunk 2]
```diff
@@ -328,38 +328,94 @@ public void testMultiTenantESIndexing() throws Exception
...
+        while (retryDelete.shouldRetry()) {
+            try {
+                boolean didElasticAckDelete = client.admin().indices().delete(
+                        new DeleteIndexRequest(indices)).get().isAcknowledged();
 
-                //  Expecting a single hit.
-                Assert.assertTrue(tenant2IndexHits.length == 1);
+                if (didElasticAckDelete) {
+                    // If Elastic acknowledged our delete wait a second to allow it time to delete the indices
+                    Thread.sleep(1000);
```
Comment (mulhollandc) [inline caf-audit/src/test/java/com/hpe/caf/auditing/elastic/ElasticAuditIT.java]: Never got an answer the last time. Why is the sleep necessary?

File: caf-audit/src/test/java/com/hpe/caf/auditing/elastic/ElasticAuditIT.java [hunk 3]
```diff
@@ -328,38 +328,94 @@ public void testMultiTenantESIndexing() throws Exception
...
+
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            } catch (ExecutionException e) {
+                throw new RuntimeException(e);
             }
         }
     }
 
+    private static SearchHit[] searchDocumentInIndices(TransportClient client, String[] indices, String field,
```
Comment (mulhollandc) [inline caf-audit/src/test/java/com/hpe/caf/auditing/elastic/ElasticAuditIT.java]: IMO I'd rename this method to searchDocument.

File: caf-audit/src/test/java/com/hpe/caf/auditing/elastic/ElasticAuditIT.java [hunk 4]
```diff
@@ -328,38 +328,94 @@ public void testMultiTenantESIndexing() throws Exception
...
+                retrySearch.retryNeeded();
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        return hits;
+    }
+
     private static SearchHit[] searchDocumentInIndex(TransportClient client, String index, String field, String value)
```
Comment (mulhollandc) [inline caf-audit/src/test/java/com/hpe/caf/auditing/elastic/ElasticAuditIT.java]: Drop this method, i.e. searchDocumentInIndex and simply re-use searchDocument (as discussed above).

Comment (mulhollandc) [reply caf-audit/src/test/java/com/hpe/caf/auditing/elastic/ElasticAuditIT.java]: Also, drop 'tenant' from any methods names and args. The arg should be passing an 'indexId'.

File: caf-audit/src/test/java/com/hpe/caf/auditing/elastic/GeneratedAuditLogIT.java [hunk 5]
```diff
@@ -66,134 +67,129 @@
...
-//        this.elasticClusterName = System.getProperty("es.cluster.name", System.getenv("es.cluster.name"));
+    @Test
+    public void auditSimpleEventTest() throws Exception {
+        Date date = new Date();
+        String correlationId = getCorrelationId();
 
-        String elasticHostname = System.getProperty("docker.host.address", "localhost");
-        int elasticPort = Integer.parseInt(System.getProperty("elasticsearch.transport.port", "9300"));
-        String elasticClusterName = System.getProperty("es.cluster.name", "elasticsearch");
+        try (AuditConnection auditConnection = AuditConnectionHelper.getAuditConnection(ES_HOSTNAME_AND_PORT,
```
Comment (andyreidz) [inline caf-audit/src/test/java/com/hpe/caf/auditing/elastic/GeneratedAuditLogIT.java]: Curious why are you now not sharing the AuditChannel and TransportClient?

**Review by andyreidz — APPROVED** [review]

**Pull request merged**